server/methods/accounts/accounts.js

```diff
@@ -398,9 +398,22 @@ export function addressBookAdd(address, accountUserId) {
 
   Meteor.users.update(Meteor.userId(), userUpdateQuery);
 
-  return Accounts.upsert({
+  const result = Accounts.upsert({
     userId
   }, accountsUpdateQuery);
+
+  // If the address update was successful, then return the full updated addrtess
+  if (result.numberAffected === 1) {
+    // Find the account
+    const updatedAccount = Accounts.findOne({
+      userId
+    });
+
+    // Pull the updated address and return it
+    return updatedAccount.profile.addressBook.find((updatedAddress) => address._id === updatedAddress._id);
+  }
+
+  return result;
```

Comment:
Need to update the jsdoc `@return`. Also, it's now more confusing to `return result` because you don't know which object structure to expect. I'm thinking just `return false` for this final result, as long as our current client calls will handle that properly.

Reply:
If we return false, then anywhere that looks for `if(error)` will not trigger, and anywhere that looks for `if(result)` will do nothing, leaving us in some nebulous state. It should probably just throw an error if it gets that far.

Reply:
@aldeed apologies if I'm misunderstanding this comment. are you suggesting to return `false` in place of the updated address result? We do need to return the updated address in GraphQL and, more generally, return values for *all* operations. That was a decision made early in the  the first round of API design.

Reply:
I've decided to throw an error, instead of returning `false` in the method. That seems to be handled well by the resolver. (Unless you wanted it the return false here for other reasons?)

Reply:
👍 Regarding my comment above, Eric explained on our standup that this is a condition where the upsert didn't change anything for some reason and would be a rare case. Agree that we shouldn't return the Mongo response there. That the address is returned in the happy path which was my only concern. Thanks!